### ISSUE TITLE
Excellent work (`mae.ipynb`)! 

### ISSUE THREAD

**Issue by sayakpaul** [ariG23498/mae-scalable-vision-learners#1]
@ariG23498 this is fantastic stuff. Super clean, readable, and coherent with the original implementation. A couple of suggestions that would likely make things even better:

* Since you have already implemented masking visualization utilities how about making them part of the `PatchEncoder` itself? That way you could let it accept a test image, apply random masking, and plot it just like the way you are doing in the earlier cells. This way I believe the notebook will be cleaner. 
* AdamW (`tfa.optimizers.adamw`) is a better choice when it comes to training Transformer-based models. 
* Are we taking the loss on the correct component? I remember you mentioning it being dealt with differently. 

After these points are addressed I will take a crack at porting the training loop to TPUs along with other performance monitoring callbacks.
